include/zephyr/drivers/stepper.h

```diff
@@ -0,0 +1,328 @@
+ */
+typedef int32_t (*stepper_move_t)(const struct device *dev, int32_t steps);
+
+/**
+ * @brief Set the max velocity in steps per seconds.
+ *
+ * See stepper_set_max_velocity() for details.
+ */
+typedef int32_t (*stepper_set_max_velocity_t)(const struct device *dev, uint32_t steps_per_second);
+
+/**
+ * @brief Set the micro-step resolution
+ *
+ * See stepper_set_micro_step_res() for details.
+ */
+typedef int32_t (*stepper_set_micro_step_res_t)(const struct device *dev,
+						enum micro_step_resolution resolution);
+
+/**
+ * @brief Set the actual a.k.a reference position of the stepper
+ *
+ * See stepper_set_actual_position() for details.
+ */
+typedef int32_t (*stepper_set_actual_position_t)(const struct device *dev, const int32_t value);
+
+/**
+ * @brief Get the actual a.k.a reference position of the stepper
+ *
+ * See stepper_get_actual_position() for details.
+ */
+typedef int32_t (*stepper_get_actual_position_t)(const struct device *dev, int32_t *value);
+
+/**
+ * @brief Set the absolute target position of the stepper
+ *
+ * See stepper_set_target_position() for details.
+ */
+typedef int32_t (*stepper_set_target_position_t)(const struct device *dev, const int32_t value);
+
+/**
+ * @brief Is the target position fo the stepper reached
+ *
+ * See stepper_is_moving() for details.
+ */
+typedef int32_t (*stepper_is_moving_t)(const struct device *dev, bool *is_moving);
+
+/**
+ * @brief Enable constant velocity mode for the stepper with a given velocity
+ *
+ * See stepper_enable_constant_velocity_mode() for details.
+ */
+typedef int32_t (*stepper_enable_constant_velocity_mode_t)(const struct device *dev,
+							   const enum stepper_direction direction,
+							   const uint32_t value);
+
+/**
+ * @brief Stepper Motor Controller API
+ */
+__subsystem struct stepper_api {
+	stepper_enable_t enable;
+	stepper_move_t move;
+	stepper_set_max_velocity_t set_max_velocity;
+	stepper_set_micro_step_res_t set_micro_step_res;
+	stepper_set_actual_position_t set_actual_position;
+	stepper_get_actual_position_t get_actual_position;
+	stepper_set_target_position_t set_target_position;
+	stepper_is_moving_t is_moving;
+	stepper_enable_constant_velocity_mode_t enable_constant_velocity_mode;
+};
+
+/**
+ * @brief Enable or Disable Motor Controller
+ * @param dev pointer to the stepper motor controller instance
+ * @param enable Input enable or disable motor controller
+ * @retval <0 Error during Enabling
+ * @retval  0 Success
+ */
+__syscall int32_t stepper_enable(const struct device *dev, bool enable);
+
+static inline int32_t z_impl_stepper_enable(const struct device *dev, bool enable)
+{
+	const struct stepper_api *api = (const struct stepper_api *)dev->api;
+
+	return api->enable(dev, enable);
+}
+
+/**
+ * @brief Set the steps to be moved from the current position i.e. relative movement
+ * @param dev pointer to the stepper motor controller instance
+ * @param steps target steps to be moved from the current position
```

Comment:
Let's use microsteps :) no conversions needed in the driver, and if whole steps are required, that can be configured by setting the microstep res